File: api/app/utils/marketplace_categories.py
from __future__ import annotations

MARKETPLACE_CATEGORIES: dict[str, list[str]] = {
    "all": [],
    "health-wellness": ["health", "wellness", "dental", "clinic", "medical", "therapy"],
    "home-services": ["home", "clean", "repair", "plumb", "electric", "maintenance"],
    "tutors-classes": ["tutor", "lesson", "class", "language", "math", "education"],
    "coaching-courses": ["coach", "course", "training", "mentor", "program"],
    "beauty-salon": ["beauty", "salon", "hair", "spa", "nail", "skin"],
    "events-photography": ["event", "photo", "wedding", "portrait", "studio"],
    "design-creative": ["design", "creative", "brand", "logo", "portfolio", "art"],
    "automotive": ["auto", "car", "vehicle", "mechanic", "garage"],
}


def category_keywords(category: str | None) -> list[str]:
    if not category or category == "all":
        return []
    return MARKETPLACE_CATEGORIES.get(category, [])

File: web/src/data/marketplaceCategories.ts
export type MarketplaceCategoryId =
  | "all"
  | "health-wellness"
  | "home-services"
  | "tutors-classes"
  | "coaching-courses"
  | "beauty-salon"
  | "events-photography"
  | "design-creative"
  | "automotive";

export type MarketplaceCategory = {
  id: MarketplaceCategoryId;
  label: string;
  icon: string;
};

export const MARKETPLACE_CATEGORIES: MarketplaceCategory[] = [
  { id: "all", label: "All Services", icon: "◆" },
  { id: "health-wellness", label: "Health & Wellness", icon: "♡" },
  { id: "home-services", label: "Home Services", icon: "⌂" },
  { id: "tutors-classes", label: "Tutors & Classes", icon: "✎" },
  { id: "coaching-courses", label: "Coaching & Courses", icon: "◎" },
  { id: "beauty-salon", label: "Beauty & Salon", icon: "✿" },
  { id: "events-photography", label: "Events & Photography", icon: "📷" },
  { id: "design-creative", label: "Design & Creative", icon: "◈" },
  { id: "automotive", label: "Automotive", icon: "⚙" },
];

export const MARKETPLACE_SORT_OPTIONS = [
  { value: "popular", label: "Popular" },
  { value: "rating", label: "Highest rated" },
  { value: "reviews", label: "Most reviewed" },
  { value: "newest", label: "Newest" },
  { value: "bookable", label: "Bookable first" },
  { value: "name", label: "Name A-Z" },
] as const;

export type MarketplaceSort = (typeof MARKETPLACE_SORT_OPTIONS)[number]["value"];

export const MARKETPLACE_RATING_OPTIONS = [
  { value: "", label: "Any rating" },
  { value: "4", label: "4.0+" },
  { value: "4.5", label: "4.5+" },
] as const;
